### PR TITLE
Allows both TLS and SSL to be specified

### PR DESCRIPTION
**Pull request by bznein**
This is needed in AppDB works when performing a TLS upgrade from an Agent that understands only SSL to one that understands TLS

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: pkg/automationconfig/automation_config_builder.go [hunk 1]
```diff
@@ -274,17 +274,11 @@ func (b *Builder) Build() (AutomationConfig, error) {
 		},
 	}
 
-	if b.tlsConfig != nil && b.sslConfig != nil {
```
Comment (rodrigovalin) [inline pkg/automationconfig/automation_config_builder.go:277]: But this will also allow for both SSL and TLS to be defined, is that correct?

Comment (bznein) [reply pkg/automationconfig/automation_config_builder.go:277]: yes that's everything this change does

**Review by chatton — APPROVED** [review]
LGTM now that the tests are passing 👍

**Review by rodrigovalin — APPROVED** [review]
LGTM

**Comment by tibulca** [comment]
LGTM

**Review by tibulca — APPROVED** [review]

**Pull request merged**